api/src/main/java/com/intern/api/client/DataBaseClient.java

```diff
@@ -49,19 +50,30 @@ public Document deleteMessage(Long messageId) {
     return collection.find(eq("from.id", userId)).limit(limitAmount).into(new ArrayList<>());
   }
 
-  public Document getUserInfo(String username) {
-    return collection.find(eq("from.username", username)).first().get("from", Document.class);
+  public Document getUserInfo(String username, Optional<Integer> charId) {
+    Document result;
+    if (charId.isPresent()) {
+      result =
+          collection
+              .find(and(eq("from.username", username), eq("chat.id", charId.get())))
+              .first()
+              .get("from", Document.class);
+    } else {
+      result = collection.find(eq("from.username", username)).first().get("from", Document.class);
+    }
+    return result;
   }
 
   public List<Document> getTopUsersByMessageAmount(int N, Optional<Integer> chatId) {
```

Comment:
It's not a good practice to name any class, variable, field, argument with one letter.